src/main/java/org/apache/commons/crypto/cipher/Openssl.java

```diff
@@ -323,11 +622,42 @@ public void clean() {
         }
     }
 
+    /**
+     * Continues a multi-part update of the Additional Authentication
+     * Data (AAD).
+     * <p>
+     * Calls to this method provide AAD to the cipher when operating in
+     * modes such as AEAD (GCM).  If this cipher is operating in
+     * either GCM mode, all AAD must be supplied before beginning
+     * operations on the ciphertext (via the {@code update} and
+     * {@code doFinal} methods).
+     *
+     * @param aad the buffer containing the Additional Authentication Data
+     *
+     */
+    public void updateAAD(byte[] aad) {
+        // must be called after initialized.
+        aadBuffer = aad.clone();
+    }
+
     /** Checks whether context is initialized. */
     private void checkState() {
         Utils.checkState(context != 0);
     }
 
+    private void processAAD() {
```

Comment:
processAAD is little strange. It was called in every call of update or doFinal.

Does Java do in the similar way? Shall this be done when updateAAD is called?